torch/futures/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""
+The ``torch.futures`` package contains a ``Future`` type and corresponding
+utility functions.
+"""
+import torch
+
+
+class Future(torch._C.Future):
+    r"""
+    Wrapper around a ``torch._C.Future``.
+    """
+    def __new__(cls):
+        return super(Future, cls).__new__(cls)
+
+    def wait(self):
```

Comment:
Nit: Should we expose `value()`/`constValue()` as well?

Reply:
Let me add a `result()` API to it.